### DISCUSSION TITLE
我实现了一个手动（enter 键）切换到下一个单词的功能

### DISCUSSION THREAD

**Discussion by tangshimin** [RealKai42/qwerty-learner#175] [Ideas]
我主要使用默写模式记忆单词，不用默写模式的话，眼睛与手指协调的好，大脑还没反应过来就自动切换到下一个单词了。使用默写单词模式的时候，耳朵听见语音之后，大脑还有一个分析的过程。

所以这个功能，主要解决我在使用默写模式记忆单词的时候遇到的如下问题：
- 一些比较长的单词和语音-字母不一一对应的词，比如 negotiate 这个词，不可能一次输入就记住了，如果想要多输入几次加强记忆，就只有输入到倒数第二个字母然后在大脑里脑补最后一个字母，然后再删掉重新输入到倒数第二个字母。觉得记得差不多了才输入最后一个字母。
- 还有一些比较短的词，虽然不长，语音和字母也是对应的，但是太短，手指已经输入完成，大脑还没有把【单词-语音-意义】建立联系，就自动的切换到下一个单词了。如果还想记忆就只有从头开始了。

手动模式的主要操作：
- 当前单词输入一遍后自动清空当前单词，等待用户重新输入。correct 计数器加1。
- 如果输入错误，wrong 计数器加1。
- 如果觉得记得差不多了就按 Enter 键切换到下一个单词

这个功能的大部分代码我已经实现了，如果大家需要这个功能的话，我会尽快提交 PR。

功能截图如下

![manual mod feature](https://user-images.githubusercontent.com/16540656/147909662-f833001e-9864-414f-8635-71947a094d8d.png)

**Comment by RealKai42** [comment]
我觉得这个思路非常酷！
目前的设计方案单词与中文含义出现的时间确实较短，很难让用户完整记忆拼写与中文含义。

但这种方式对用户之前使用软件的习惯改变较大，或许可以先 提交一个 draft pr 让有兴趣的人体验一下，然后我们一起讨论一下如何平滑地融入到现有的网站中

**Comment by tangshimin** [comment]
之前的使用习惯并不会影响，我把之前的习惯设置成自动模式，可以用 ctrl+X 切换。
可以先把自动模式设置成默认模式

![manual mod feature-1](https://user-images.githubusercontent.com/16540656/147947122-57b56b21-4fda-4284-90fc-caa83821eedd.png)
